Visual Studio 2022 fails to compile MAMe due to broken overload resolution

### MAME version

5599230d9364847a25e836dbd2df34fa6f936d4d

### System information

Windows Server 2022 Datacenter 21H2 amd64

### INI configuration details

_No response_

### Emulated system/software

Visual Studio 2022 17.5.5

### Incorrect behaviour

```
F:\gitP\mamedev\mame\src\lib\util\strformat.h(1144,15): error C2039: 'empty': is not a member of 'emu_fatalerror' [F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019\optional.vcxproj]
F:\gitP\mamedev\mame\src\lib\util\strformat.h(1144,15): error C2039: 		, m_end(fmt.empty() ? nullptr : (m_begin + std::distance(std::cbegin(fmt), std::cend(fmt)))) [F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019\optional.vcxproj]
F:\gitP\mamedev\mame\src\lib\util\strformat.h(1144,15): error C2039: 		            ^ [F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019\optional.vcxproj]
F:\gitP\mamedev\mame\src\lib\util\strformat.h(1144,65): error C2672: 'std::cbegin': no matching overloaded function found [F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019\optional.vcxproj]
F:\gitP\mamedev\mame\src\lib\util\strformat.h(1144,65): error C2672: 		, m_end(fmt.empty() ? nullptr : (m_begin + std::distance(std::cbegin(fmt), std::cend(fmt)))) [F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019\optional.vcxproj]
F:\gitP\mamedev\mame\src\lib\util\strformat.h(1144,65): error C2672: 		                                                              ^ [F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019\optional.vcxproj]
F:\gitP\mamedev\mame\src\lib\util\strformat.h(1144,83): error C2672: 'std::cend': no matching overloaded function found [F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019\optional.vcxproj]
```

### Expected behaviour

build successfully.

### Steps to reproduce

1. git clone https://github.com/mamedev/mame F:\gitP\mamedev\mame
2. git -C "F:\gitP\mamedev\mame" submodule sync
3. git -C "F:\gitP\mamedev\mame" submodule update --init --recursive
4. cd F:\gitP\mamedev\mame
5. set PATH=F:\gitP\mamedev\mame\..\tools\msys64\usr\bin;F:\gitP\mamedev\mame\..\tools\msys64\mingw64\bin;%PATH%
6. make SUBTARGET=tiny PTR64=1 TOOLS=1 OPTIMIZE=0 NOWERROR=1 MODERN_WIN_API=1 NO_USE_PORTAUDIO=1 vs2019 -j4 2>&1
7. cd F:\gitP\mamedev\mame\build\projects\windows\mametiny\vs2019
8. devenv /upgrade mametiny.sln 2>&1
9. msbuild /m /p:Platform=x64 /p:Configuration=Release /p:WindowsTargetPlatformVersion=10.0.22621.0 /p:PreferredToolArchitecture=x64 mametiny.sln /t:Rebuild

### Additional details

detailed log: [build.log](https://github.com/mamedev/mame/files/11659332/build.log)

--- a/src/emu/emucore.h
+++ b/src/emu/emucore.h
@@ -233,7 +233,7 @@ public:
 	emu_fatalerror(util::format_argument_pack<char> const &args);
 	emu_fatalerror(int _exitcode, util::format_argument_pack<char> const &args);
 
-	template <typename Format, typename... Params>
+	template <typename Format, typename... Params, typename = std::enable_if_t<!std::is_base_of_v<emu_fatalerror, std::remove_reference_t<Format>>>>
 	emu_fatalerror(Format &&fmt, Params &&... args)
 		: emu_fatalerror(static_cast<util::format_argument_pack<char> const &>(util::make_format_argument_pack(std::forward<Format>(fmt), std::forward<Params>(args)...)))
 	{

